File: leetcode/code/292.nim_game.py
class Solution(object):
    def canWinNim(self, n):
        return n % 4 != 0


if __name__ == "__main__":
    for i in [1, 2, 3, 5, 6, 7, 9]:
        assert Solution().canWinNim(i) is True

    for i in [4, 8, 12]:
        assert Solution().canWinNim(i) is False

File: leetcode/code/292.nim_game.go
package main

func canWinNim(n int) bool {
    return n % 4 != 0
}

func main() {
    if !(canWinNim(1) && canWinNim(2) && canWinNim(3) && canWinNim(5)) {
        println("buggy")
    }
    if canWinNim(4) && canWinNim(8) {
        println("buggy")
    }
}
